lib/pbench/server/indexing_tarballs.py

```diff
@@ -517,3 +535,35 @@ def sigquit_handler(*args):
                     pass
 
         return res.value
+
+    @staticmethod
+    def error_code():
+        error_code = Errors(
+            ErrorCode("OK", 0, None, "Successful completion"),
+            ErrorCode("OP_ERROR", 1, False, "Operational error while indexing"),
+            ErrorCode("CFG_ERROR", 2, False, 'Configuration file not specified"'),
+            ErrorCode("BAD_CFG", 3, False, "Bad configuration file"),
+            ErrorCode(
+                "TB_META_ABSENT",
+                4,
+                True,
+                "Tar ball does not contain a metadata.log file",
+            ),
+            ErrorCode("BAD_DATE", 5, True, 'Bad start run date value encountered"'),
+            ErrorCode("FILE_NOT_FOUND_ERROR", 6, True, "File Not Found error"),
+            ErrorCode("BAD_METADATA", 7, True, "Bad metadata.log file encountered"),
+            ErrorCode(
+                "MAPPING_ERROR",
+                8,
+                False,
+                "Error reading a mapping file for Elasticsearch templates",
+            ),
+            ErrorCode(
+                "TEMPLATE_CREATION_ERROR",
+                9,
+                False,
+                "Error creating one of the Elasticsearch templates",
+            ),
+            ErrorCode("GENERIC_ERROR", 12, False, "Unexpected error encountered"),
+        )
+        return error_code
```

Comment:
I think that the `error_code(s)` list should be constructed once as it was before, just within the scope of the module. (Or even as a Class property) rather than constructing it dynamically in multiple places with this method. They don't change!

Reply:
fixed